source/ui/beamline/AMCompositeControlDirectionMoveButton.h

```diff
@@ -0,0 +1,90 @@
+#ifndef AMCOMPOSITECONTROLDIRECTIONMOVEBUTTON_H
+#define AMCOMPOSITECONTROLDIRECTIONMOVEBUTTON_H
+
+#include <QToolButton>
+#include <QList>
+#include <QDialog>
+#include <QButtonGroup>
+
+class AMControl;
+class AMCompositeControlMoveButtonContextMenu;
+
+#include "AMControlMoveButton.h"
+
+class AMCompositeControlDirectionMoveButton : public AMControlMoveButton
+{
+	Q_OBJECT
+public:
+
+	/// Create a move button for the given \c control. The available move step sizes (in the control's units) are given by \c stepSizes.  If you specify \c reverseDirection, the direction of the move will be opposite of the sign of the values in stepSizes.  It's okay for \c control to be 0. This button will simply do nothing (and be disabled) until you call setControl() with a valid control.  This widget does not take ownership of the control.
+	virtual ~AMCompositeControlDirectionMoveButton(){}
+	explicit AMCompositeControlDirectionMoveButton(QWidget *parent = 0,
+								 AMControl* control = 0, AMControl* control2 = 0, AMControl* controlR = 0,
+								 QList<double> stepSizes = QList<double>() << 1 << 5 << 10,
+								 bool directionReversed = false, bool direction2Reversed = false);
+
+	/// Set the first control that is moved by this button. It's okay for \c control to be 0, in which case this button will simply do nothing until you call setControl() again with a valid control.
+	void setControl(AMControl* control);
```

Comment:
I think it is pretty universally useful.  And X-Y-Theta motor control set can be fed to it to get movements along and perpendicular to theta...   Your call though.   (It's own class and clearer control names will help make that obvious.)